lambdas/models/fhir/R4/fhir_document_reference.py

```diff
@@ -127,6 +127,19 @@ class DocumentReference(BaseModel):
     context: Optional[DocumentReferenceContext] = None
     meta: Optional[Meta] = None
 
+    def extract_nhs_number_from_fhir(self) -> str:
+        """Extract NHS number from FHIR document"""
+        # Extract NHS number from subject.identifier where the system identifier is NHS number
+        if (
+            self.subject
+            and self.subject.identifier
+            and self.subject.identifier.system
+            == "https://fhir.nhs.uk/Id/nhs-number"
+        ):
+            return self.subject.identifier.value
+        else:
+            return None
```

Comment:
Does this need to raise an error. I see the POST fhir service does. And is it possible, if we're going to duplicate methods that we share them rather than each having their own version?